packages/bitcore-node/src/providers/chain-state/internal/internal.ts

```diff
@@ -383,6 +383,9 @@ export class InternalStateProvider implements IChainStateService {
       wallets: wallet._id,
       'wallets.0': { $exists: true }
     };
+    if (wallet.chain === 'BTC' && ['testnet3', 'testnet4'].includes(wallet.network)) {
```

Comment:
You may avoid confusion if you check this conditional on the initial query assignment (line 380):

```javascript
const query = {
   ...,
   network: wallet.chain === 'BTC' && ['testnet3', 'testnet4'].includes(wallet.network) ? wallet.network : network
}
```

Reply:
I don't want to set the network to anything if the condition is not met, including undefined.